### PR TITLE
docs: Add additional documentation to aid in first-time setup

### PR DESCRIPTION
**Pull request by cmanallen**
I had some trouble running the example so I've added documentation to improve the experience for new users.

### CODE REVIEW

File: examples/transform_and_produce/README.md [hunk 1]
```diff
@@ -59,3 +59,21 @@ Produce a message to the topic by typing in a message in the expected format and
...
+   - It can be found in `sentry/src/sentry/conf/server.py`.
+4. Run `sentry devservices up`.
+5. You should see the following in stdout (or similar).
+
+```
+> Pulling image '.../cp-kafka:6.2.0'
+> Starting EXISTING container 'sentry_kafka' (listening: ('127.0.0.1', 9092))
+```
+
+Congratulations! You can now run docker commands against the `sentry_kafka` container. If you are still having issues reach out on the `#discuss-eng-sns` Slack channel.
```
Comment (onewland) [inline examples/transform_and_produce/README.md:79]: I think we should drop this last sentence -- presumably this library is/could be used by people outside Sentry.

**Review by onewland — COMMENTED** [review]
This is kind of a tough call because we don't want `arroyo` to be tightly coupled to Sentry itself -- it's supposed to be a general-purpose streaming processing library. Though this example already assumes that you're running the sentry application, it doesn't mean we should continue down that path

Maybe we could change the overall example README to run a known Kafka docker image to remove the dependency on `sentry_kafka`?

**Comment by lynnagara** [comment]
Agree with @onewland here. I don't think these details are relevant for this library. All that this library's tests and examples require is a running Kafka on 127.0.0.1:9092. I think we may want to include some configuration here to spin these up instead of depending on Sentry's.

**Comment by cmanallen** [comment]
@lynnagara @onewland Because running the example requires service orchestration (and independence from the sentry project) I think a tool like docker-compose would be beneficial.  Would that change be welcome?

**Comment by lynnagara** [comment]
@cmanallen That seems good to me. If we can share the same configuration with CI https://github.com/getsentry/arroyo/blob/main/.github/workflows/ci.yml#L59-L76 that would be great.

**Pull request closed without merging**